lib/ruby_lsp/requests/completion.rb

```diff
@@ -63,7 +65,7 @@ def on_string_node_enter(node)
       # Handle completion on regular constant references (e.g. `Bar`)
       sig { params(node: Prism::ConstantReadNode).void }
       def on_constant_read_node_enter(node)
-        return if DependencyDetector.instance.typechecker
+        return if @typechecker_enabled
```

Comment:
Oh nice yeah I mean if sorbet can cover this then that works! I had sorta considered that but it was way easier for me to understand how to change this code base.

Will we get 1:1 support for `typed: false` from the sorbet LS as this one though? 

I know constants will be covered but I think you all are working on more support for methods in here as well right? Still seems like it could be valuable to know at the document level what's going on with typechecking.

Obviously this is kind of an edge case - It's on my roadmap to go get us past `typed: false` but it may take some time still. We need to dedicate some time to supporting that push properly and have a few other priorities before we can take the time to do that. We kinda yolo'd into it on first pass and it was messy / got in the way of devs quite a bit so we had to regress for a time. Hence why we have sorbet installed, but are largely `typed: false` for now. 

Anywhoo context aside I still think this could be valuable but I'm also happy to wait for a concrete use-case if the current one gets covered by sorbet!

Reply:
Yeah, this change still makes perfect sense for methods. Sorbet does not even attempt to typecheck `typed: false` files, so we can definitely fallback method hover, completion, signature help and definition to the Ruby LSP.

For constants though, we can fully delegate to Sorbet. It's always able to discover and handle constants properly (meta-programming excluded of course).

Reply:
Right yeah that makes sense. I just pushed those changes up now 👍